src/burnchains/burnchain.rs

```diff
@@ -157,14 +160,55 @@ impl BurnchainStateTransition {
         let consumed_leader_keys =
             sort_tx.get_consumed_leader_keys(&parent_snapshot, &block_commits)?;
 
+        // assemble the commit windows
+        let mut windowed_block_commits = vec![block_commits];
+        let mut windowed_user_burns = vec![user_burns];
+
+        for blocks_back in 0..(MINING_COMMITMENT_WINDOW - 1) {
+            if parent_snapshot.block_height < (blocks_back as u64) {
+                debug!("Mining commitment window shortened because block height is less than window size";
+                       "block_height" => %parent_snapshot.block_height,
+                       "window_size" => %MINING_COMMITMENT_WINDOW);
+                break;
+            }
+            let block_height = parent_snapshot.block_height - (blocks_back as u64);
+            let sortition_id = match sort_tx.get_block_snapshot_by_height(block_height)? {
```

Comment:
It might be more efficient to just read the parent snapshot using the current snapshot's parent snapshot pointer (this would save you a MARF read per block height).

Reply:
The snapshot only stores a burnchain header hash, which would need a MARF lookup to resolve to a SortitionId. We could extend the BlockSnapshot structure to store a parent pointer as SortitionId, though.

Reply:
I thought the BlockSnapshot already stored the sortition ID?  If not, then don't worry about it in this PR.